src/HttpHelper/HttpClientHelper/HttpClient.php

```diff
@@ -7,80 +7,121 @@
-        $request && $aLogMetadata['request'] = $request->getBody();
-
-        if (!$response->isSuccess()) {
-            Logger::error("$sAction has failed: {$response->getErrorMessage()}", $logContext, $aLogMetadata);
-        } else {
-            Logger::debug("$sAction succeed", $logContext, $aLogMetadata);
-        }
+        return $this->request(Request::METHOD_PATCH, $sUrl, $aRequestOptions, $sAction, $logContext, $aLogMetadata);
     }
 
-    private function request(string $sMethod, string $sUrl, array $aRequestOptions, $fRequestTimeout = null): HttpClientResponse
+    private function request(
+        string $sMethod,
+        string $sUrl,
+        array $aRequestOptions,
+        string $sAction,
+        LogContext $logContext,
+        array $aLogMetadata = []
+    ): ResponseInterface
     {
-        $bError = true;
         try {
-            $aRequestOptions[RequestOptions::TIMEOUT] = $fRequestTimeout ? : self::DEFAULT_REQUEST_TIMEOUT;
+            $aRequestOptions[RequestOptions::TIMEOUT] = $aRequestOptions[RequestOptions::TIMEOUT] ?? self::DEFAULT_REQUEST_TIMEOUT;
             $response = $this->httpClient->request($sMethod, $sUrl, $aRequestOptions);
+
             $sResponseBody = $response->getBody()->getContents();
-            $iHttpCode = $response->getStatusCode();
-            $bError = false;
-        } catch (ConnectException $e) {
-            $sResponseBody = "ConnectException: {$e->getMessage()}";
-            $iHttpCode = 0;
+            $response->getBody()->rewind();
+            Logger::debug("$sAction : succeed requesting $sMethod $sUrl - {$response->getStatusCode()}", $logContext, \array_merge($aLogMetadata, [
+                'response' => $sResponseBody,
+                'requestOptions' => $this->hideRequestSecrets($aRequestOptions)
+            ]));
         } catch (GuzzleException $e) {
             if (\method_exists($e, 'hasResponse') && $e->hasResponse()) {
                 $response = $e->getResponse();
-                $sResponseBody = $response->getBody()->getContents();
+                $sMessage = $response->getBody()->getContents();
                 $iHttpCode = $response->getStatusCode();
             } else {
-                $sResponseBody = "GuzzleException: {$e->getMessage()}";
-                $iHttpCode = 0;
+                $sMessage = $e->getMessage();
+                $iHttpCode = $e->getCode();
             }
+
+            Logger::error("$sAction : error while requesting $sMethod $sUrl - $iHttpCode", $logContext, \array_merge($aLogMetadata, [
+                'response' => $sMessage,
+                'requestOptions' => $this->hideRequestSecrets($aRequestOptions)
+            ]));
+
+            throw new HttpClientException($sMessage, $iHttpCode);
+
+        } catch (\Throwable $e) {
+            Logger::exception("$sAction : error while requesting $sMethod $sUrl", $logContext, $e, $aLogMetadata);
+
+            throw $e;
+        }
+
+        return $response;
```

Comment:
I have a question about this. Before in pl3 for exemple we always worked with something like:
`return new HttpClientResponse($sResponseBody, $iHttpCode);`
But now we are going to have just a response, which is fine. 
So how it's going to feet with previous request ? Do we have to adapt the code in pl3 for exemple ?
Maybe there is some kind of response that you added but I don't see it.

Reply:
Yep no choice, we will have to adapt the code in PL3.